diff --git a/node_modules/@airgap/beacon-core/dist/cjs/storage/LocalStorage.js b/node_modules/@airgap/beacon-core/dist/cjs/storage/LocalStorage.js
index 4542fb1..6118314 100644
--- a/node_modules/@airgap/beacon-core/dist/cjs/storage/LocalStorage.js
+++ b/node_modules/@airgap/beacon-core/dist/cjs/storage/LocalStorage.js
@@ -58,21 +58,23 @@ var LocalStorage = /** @class */ (function () {
         return __awaiter(this, void 0, void 0, function () {
             var value;
             return __generator(this, function (_a) {
-                value = localStorage.getItem(this.getPrefixedKey(key));
-                if (!value) {
-                    if (typeof beacon_types_1.defaultValues[key] === 'object') {
-                        return [2 /*return*/, JSON.parse(JSON.stringify(beacon_types_1.defaultValues[key]))];
+                if (typeof localStorage !== 'undefined') {
+                    value = localStorage.getItem(this.getPrefixedKey(key));
+                    if (!value) {
+                        if (typeof beacon_types_1.defaultValues[key] === 'object') {
+                            return [2 /*return*/, JSON.parse(JSON.stringify(beacon_types_1.defaultValues[key]))];
+                        }
+                        else {
+                            return [2 /*return*/, beacon_types_1.defaultValues[key]];
+                        }
                     }
                     else {
-                        return [2 /*return*/, beacon_types_1.defaultValues[key]];
-                    }
-                }
-                else {
-                    try {
-                        return [2 /*return*/, JSON.parse(value)];
-                    }
-                    catch (jsonParseError) {
-                        return [2 /*return*/, value]; // TODO: Validate storage
+                        try {
+                            return [2 /*return*/, JSON.parse(value)];
+                        }
+                        catch (jsonParseError) {
+                            return [2 /*return*/, value]; // TODO: Validate storage
+                        }
                     }
                 }
                 return [2 /*return*/];
@@ -82,11 +84,13 @@ var LocalStorage = /** @class */ (function () {
     LocalStorage.prototype.set = function (key, value) {
         return __awaiter(this, void 0, void 0, function () {
             return __generator(this, function (_a) {
-                if (typeof value === 'string') {
-                    return [2 /*return*/, localStorage.setItem(this.getPrefixedKey(key), value)];
-                }
-                else {
-                    return [2 /*return*/, localStorage.setItem(this.getPrefixedKey(key), JSON.stringify(value))];
+                if (typeof localStorage !== 'undefined') {
+                    if (typeof value === 'string') {
+                        return [2 /*return*/, localStorage.setItem(this.getPrefixedKey(key), value)];
+                    }
+                    else {
+                        return [2 /*return*/, localStorage.setItem(this.getPrefixedKey(key), JSON.stringify(value))];
+                    }
                 }
                 return [2 /*return*/];
             });
